pymc3/distributions/distribution.py

```diff
@@ -219,9 +227,15 @@ def __new__(
             # A batch size was specified through `dims`, or implied by `observed`.
             rv_out = change_rv_size(rv_var=rv_out, new_size=resize_shape, expand=True)
 
-        if initval is not None:
+        if initval is not UNSET and initval is not None:
             # Assigning the testval earlier causes trouble because the RV may not be created with the final shape already.
             rv_out.tag.test_value = initval
+        elif initval is None and hasattr(rv_out.tag, "test_value"):
+            warnings.warn(
+                f"An initval=None was specified, but the {cls.__name__} distribution assigned a test value anyways."
+                f" This unexpected behaviour and should be investigated!",
```

Comment:
Are we sure that there aren't any Ops doing this legally?

Reply:
I am not sure, but if having an initval automatically set upstream changes the behavior in PyMC3 wouldn't it be better to find out ASAP (ie tests failing / users opening an issue)?

Reply:
Your're right, we should fail faster.
Let's add it as another commit at the end after making the tests pass first. There are still some unexplained failures =/

Reply:
The `Flat` distributions assign a test value and I wouldn't be surprised if bad things happened if we were to remove that (because they can't draw random samples).
How should we proceed w.r.t. this warning then?

Reply:
Either remove the warning or add an exception for the distributions that must assign one. Neither option sounds good to me, but maybe removing the warning is okay?